src/coprocessor/dag/expr/builtin_math.rs

```diff
@@ -308,6 +330,50 @@ mod test {
         }
     }
 
+    #[test]
+    fn test_log2() {
+        let tests = vec![
+            (ScalarFuncSig::Log2, Datum::F64(16f64), Datum::F64(4f64)),
+            (
+                ScalarFuncSig::Log2,
+                Datum::F64(5f64),
+                Datum::F64(2.321928094887362f64),
+            ),
+            (ScalarFuncSig::Log2, Datum::F64(-1.234f64), Datum::Null),
+            (ScalarFuncSig::Log2, Datum::F64(0f64), Datum::Null),
+        ];
+        let mut ctx = EvalContext::default();
+        for (sig, arg, exp) in tests {
+            let arg = datum_expr(arg);
+            let f = scalar_func_expr(sig, &[arg]);
+            let op = Expression::build(&mut ctx, f).unwrap();
+            let got = op.eval(&mut ctx, &[]).unwrap();
+            assert_eq!(got, exp);
+        }
+    }
+
+    #[test]
+    fn test_log10() {
+        let tests = vec![
+            (ScalarFuncSig::Log10, Datum::F64(100f64), Datum::F64(2f64)),
+            (
+                ScalarFuncSig::Log10,
+                Datum::F64(101f64),
+                Datum::F64(2.0043213737826426f64),
```

Comment:
@sllt  We had some discussion of whether to use a specific f64 number, as different architect might give slight different result, making the test fails.

You can see more discussion here: https://github.com/tikv/tikv/pull/3406

I think it is safe to just use `assert!((got.f64() - exp).abs() < 1e-10_f64);` to do comparison, instead of doing the strong == on f64.

You can see the official Rust f64 doc is doing the same:
https://doc.rust-lang.org/std/primitive.f64.html#method.log2

Reply:
Any updates?